D-lama-service/Data/ProjectEntities/DataPoint.cs

```diff
@@ -10,12 +10,23 @@ public abstract class DataPoint : Entity
     {
         [Required]
         public int DataPointIndex { get; set; }
+
         [Required]
         [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
         public DateTime CreationDate { get; set; } // on create
+
         [Required]
         public DateTime UpdateDate { get; set; } // on update
+
         [Required]
         public int Version { get; set; } // increment per update
+
+        // Required foreign key property
+        public int ProjectId { get; set; }
```

Comment:
Good idea to make the FK refernce in DataPoint already. Could you please then delete the navigation in TextDataPoint? It is not needed there now (remove lines 14-19 in file TextDataPoint.cs)